File: src/infrastructure/entities/Product_Tag.entity.ts
import { Entity, PrimaryGeneratedColumn, ManyToOne, JoinColumn } from "typeorm";

import ProductEntity from "./Product.entity";
import TagEntity from "./Tag.entity";

@Entity("product_tags")
export default class ProductTagEntity {
  @PrimaryGeneratedColumn("increment", { type: "bigint" })
  id: number;

  @ManyToOne(() => ProductEntity, (product) => product.id, {
    onDelete: "CASCADE",
  })
  @JoinColumn({ name: "product_id" })
  product: ProductEntity;

  @ManyToOne(() => TagEntity, (tag) => tag.id, { onDelete: "CASCADE" })
  @JoinColumn({ name: "tag_id" })
  tag: TagEntity;
}

File: src/main/java/com/example/demo/product/entity/ProductTagEntity.java
package com.example.demo.product.entity;

import com.example.demo.tag.entity.TagEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Table(name = "product_tags")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class ProductTagEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
    @JoinColumn(name = "product_id")
    private ProductEntity productEntity;
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
    @JoinColumn(name = "tag_id")
    private TagEntity tagEntity;

    public static ProductTagEntity create(
            ProductEntity productEntity,
            TagEntity tagEntity
    ) {
        return new ProductTagEntity(
                null,
                productEntity,
                tagEntity
        );
    }
}
